CONTRIBUTING.md

```diff
@@ -0,0 +1,69 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# How to contribute to Apache Arrow Go
+
+## Did you find a bug?
+
+The Arrow project uses GitHub as a bug tracker.  To report a bug, sign in to
+your GitHub account, navigate to [GitHub issues](https://github.com/apache/arrow-go/issues)
+and click on **New issue** .
+
+To be assigned to an issue, add a comment "take" to that issue.
```

Comment:
We don't have this feature in apache/arrow-go.
Do we need this? If we need this, we need to port https://github.com/apache/arrow/blob/0c3206772051c52fe2e648623daaca2c4e71cb72/.github/workflows/comment_bot.yml#L178-L194 to apache/arrow-go.

Reply:
I'd be more than willing to switch arrow-go to instead use conventional commits like ADBC does, what do you think @kou?

Reply:
I'd be able to update the automation to have the semantic versioning automatically handled via those commit messages

Reply:
I think it might be a good idea for us to port that, it's likely useful

Reply:
OK. Let's port it: GH-168